perf/benchmark/security/generate_policies/generate_policies.go

```diff
@@ -80,180 +101,177 @@ func PolicyToYAML(policy *MyPolicy, spec proto.Message) (string, error) {
 	spec.Rules = ruleList
 
-	yaml, err := PolicyToYAML(policy, spec)
+	yaml, err := PolicyToYAML(policyHeader, spec)
 	if err != nil {
 		return "", err
 	}
 	return yaml, nil
 }
 
-func generateRules(action string, ruleToGenerator map[string]*ruleGenerator,
-	policy *MyPolicy, ruleMap map[string]int) (string, error) {
+func generatePeerAuthentication(policyData SecurityPolicy, policyHeader *MyPolicy) (string, error) {
+	spec := &authzpb.PeerAuthentication{
+		Mtls: &authzpb.PeerAuthentication_MutualTLS{},
+	}
+	switch policyData.PeerAuthN.MtlsMode {
+	case "STRICT":
+		spec.Mtls.Mode = authzpb.PeerAuthentication_MutualTLS_STRICT
+	case "DISABLE":
+		spec.Mtls.Mode = authzpb.PeerAuthentication_MutualTLS_DISABLE
+	case "":
+		spec.Mtls.Mode = authzpb.PeerAuthentication_MutualTLS_STRICT
+	default:
+		return "", fmt.Errorf("invalid mtlsMode: %s", policyData.PeerAuthN.MtlsMode)
+	}
 
-	switch policy.Kind {
+	yaml, err := PolicyToYAML(policyHeader, spec)
+	if err != nil {
+		return "", err
+	}
+	return yaml, nil
+}
+
+func generateRules(policyData SecurityPolicy, policyHeader *MyPolicy) (string, error) {
+	switch policyHeader.Kind {
 	case "AuthorizationPolicy":
-		return generateAuthorizationPolicy(action, ruleToGenerator, policy, ruleMap)
+		return generateAuthorizationPolicy(policyData, policyHeader)
 	case "PeerAuthentication":
-		return "", fmt.Errorf("unimplemented")
+		return generatePeerAuthentication(policyData, policyHeader)
 	case "RequestAuthentication":
 		return "", fmt.Errorf("unimplemented")
 	default:
-		return "", fmt.Errorf("unknown policy kind: %s", policy.Kind)
+		return "", fmt.Errorf("unknown policy kind: %s", policyHeader.Kind)
 	}
 }
 
 func createPolicyHeader(namespace string, name string, kind string) *MyPolicy {
+	if namespace == "" {
+		namespace = "twopods-istio"
+	}
 	return &MyPolicy{
 		APIVersion: "security.istio.io/v1beta1",
 		Kind:       kind,
 		Metadata:   MetadataStruct{Namespace: namespace, Name: name},
 	}
 }
 
-func createRuleGeneratorMap(ruleToOccurancesPtr map[string]int) map[string]*ruleGenerator {
-	ruleGeneratorMap := make(map[string]*ruleGenerator)
-
-	if ruleToOccurancesPtr["numSourceIP"] > 0 || ruleToOccurancesPtr["numNamespaces"] > 0 {
-		ruleGeneratorMap["from"] = &ruleGenerator{
-			gen: sourceGenerator{},
-		}
-	}
+func generatePolicy(policyData SecurityPolicy, kind string, numPolicy int, totalPolicies int) (int, error) {
+	for i := 1; i <= numPolicy; i++ {
+		testName := fmt.Sprintf("test-%s-%d", kind, i)
+		policyHeader := createPolicyHeader(policyData.Namespace, testName, kind)
 
-	if ruleToOccurancesPtr["numPaths"] > 0 {
-		ruleGeneratorMap["to"] = &ruleGenerator{
-			gen: operationGenerator{},
+		rules, err := generateRules(policyData, policyHeader)
+		if err != nil {
+			return 0, err
 		}
-	}
-
-	if ruleToOccurancesPtr["numValues"] > 0 {
-		ruleGeneratorMap["when"] = &ruleGenerator{
-			gen: conditionGenerator{},
+		yaml := bytes.Buffer{}
+		yaml.WriteString(rules)
+		if totalPolicies > 1 {
```

Comment:
why do we need `totalPolicies` policies here? shouldn't we always append the `---` after each policy

Reply:
`totalPolicies` is here to make sure we do not add a `---` on the very last policy. But this does not really make a difference since a policy can be applied with a ending `---`